markdownreveal/commands.py

```diff
@@ -84,11 +94,10 @@ def upload(markdown_file: Path):
         shell('git status')
     except CalledProcessError as error:
         return
-    origin = shell('git config remote.origin.url')[0]
-    if 'github' not in origin:
-        error = 'Uploading only supported for GitHub repositories!'
+
+    if not is_valid_remote(remote):
+        error = 'Option supplied for remote (%s) is not valid' % remote
```

Comment:
I am not convinced about this `is_valid_remote()` function and error message. I think it hides information about what went wrong from the user.

What if...

Instead of:

```python
try:
    shell('git status')
except CalledProcessError as error:
    return
```
We move the `remote_url` definition here:

```python
try:
    shell('git status')
    remote_url = shell('git remote get-url ' + remote)[0]
except CalledProcessError as error:
    return
```

This way if the remote is not valid the appropriate message will be shown to the user by the `git` command, so we do not need to write it. It will be very clear for the user what went wrong.

If we also want to check the remote URL is a valid *push* URL, we can add the `--push` option:

```python
try:
    shell('git status')
    remote_url = shell('git remote get-url --push ' + remote)[0]
except CalledProcessError as error:
    return
```

Once we have done this I think there is no need for the `is_valid_remote()` function anymore. We have already checked it is a valid push remote and, if not, we are already displaying a clear error message.

Then for the `github` URL check we can leave the code as it was written before, with the explicit message that, for now, only GitHub is supported for uploading the presentation.

What do you think? :blush: